### DISCUSSION TITLE
don't index the whole chain

### DISCUSSION THREAD

**Discussion by rajgoesout** [filecoin-project/lily#319] [Q&A] [answered]
I first indexed using `-indexhead` flag, which didn't take much time, but the `-indexhistory` seems to take a really long time given that I am indexing the nerpa devnet. I also see that some tables like `actors`, `actor_states`, `chain_visualizer_blocks_view`, etc are getting filled, but tables like `miner_infos`, `miner_sector_infos`, etc are empty. 

From what I understand, visor will first fill up all the chain data (from genesis), and after that, fill up the miner tables. Is that correct? Is it possible to index only a certain timespan (eg: last/first 5000 epochs), and fill all the tables (like miner info) for that much chain data?

**Comment by iand** [comment, marked as answer]
Hi, We recently add a new command called "walk" which should work better for you.

You can run it like this to index miner states:

```
visor walk --from 1000 --to 2000 --tasks=actorstatesminer 
```

You can include multiple tasks by comma separating them. The full list is here https://github.com/filecoin-project/sentinel-visor/blob/master/chain/indexer.go#L23
